### PR TITLE
Search user across name or email

### PR DESCRIPTION
**Pull request by phunghangbk**

### CODE REVIEW

File: app/views/admin/users/index.html.erb [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
 <h1><%= t "All_user" %></h1>
-<%= will_paginate %>
-<div class="container">
-  <table class="table table-bordered">
-    <%= render @users %>
-  </table>
-</div>
-<%= will_paginate %>
+
+<%= form_tag(admin_users_path, :method => "get", id: "search-form") do %>
```
Comment (vuducdat) [inline app/views/admin/users/index.html.erb]: should method: :get

File: app/models/user.rb [hunk 2]
```diff
@@ -12,4 +12,8 @@ class User < ActiveRecord::Base
   def set_default_role
     self.role ||= :normal
   end
+
+  def self.search(search)
```
Comment (biznezo) [inline app/models/user.rb]: try to remove ()

File: app/views/admin/users/index.html.erb [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
+<% if @users.present? %>
+  <%= will_paginate %>
+  <div class="container">
+    <table class="table table-bordered">
+      <%= render @users %>
+    </table>
+  </div>
+  <%= will_paginate %>
+<% else %>
+  <p>There are no posts containing the term(s) <%= params[:search] %>.</p>
```
Comment (vuducdat) [inline app/views/admin/users/index.html.erb]: I18n Hằng ơi. Đừng tức giận vì tớ comment nhé. :DD

File: app/models/user.rb [hunk 4]
```diff
@@ -12,4 +12,8 @@ class User < ActiveRecord::Base
...
+    where("name LIKE ? OR email LIKE ?", "%#{search}%", "%#{search}%")
```
Comment (biznezo) [inline app/models/user.rb]: name OR email ?

File: app/views/admin/users/index.html.erb [hunk 5]
```diff
@@ -1,9 +1,19 @@
...
-<div class="container">
-  <table class="table table-bordered">
-    <%= render @users %>
-  </table>
-</div>
-<%= will_paginate %>
+
+<%= form_tag(admin_users_path, method: :get, id: "search-form") do %>
+  <%= text_field_tag :search, params[:search], placeholder: "Search" %>
+  <%= submit_tag "Search" %>
```
Comment (biznezo) [inline app/views/admin/users/index.html.erb]: i18n

File: app/models/user.rb (hunk 4 above)
Comment (phunghangbk) [inline app/models/user.rb]: Nhu the thi bao loi anh a

**Comment by vuducdat** [comment]
LGTM

**Comment by biznezo** [comment]
OK

**Comment by phuongbk** [comment]
Dùng ransack đi

File: app/models/user.rb [hunk 6]
```diff
@@ -12,4 +12,8 @@ class User < ActiveRecord::Base
   def set_default_role
     self.role ||= :normal
   end
+
+  def self.search search
+    where("name LIKE ? OR email LIKE ?", "%#{search}%", "%#{search}%")
```
Comment (dieunb) [inline app/models/user.rb]: remove `()`

File: app/controllers/admin/users_controller.rb [hunk 7]
```diff
@@ -2,7 +2,13 @@ class Admin::UsersController < ApplicationController
   load_and_authorize_resource
 
   def index
-    @users = @users.paginate page: params[:page]
+    @search = User.search(params[:q])
+    @users = @search.result.page params[:page]
+
+    respond_to do |format|
+      format.html
+      format.json { render json: [:admin, @users]}
```
Comment (biznezo) [inline app/controllers/admin/users_controller.rb]: indent

File: app/views/admin/users/index.html.erb [hunk 8]
```diff
@@ -1,9 +1,23 @@
 <% provide :title, t("All_user") %>
 <h1><%= t "All_user" %></h1>
-<%= will_paginate %>
-<div class="container">
-  <table class="table table-bordered">
-    <%= render @users %>
-  </table>
-</div>
-<%= will_paginate %>
+<%= search_form_for @search, url: admin_users_path,html: {method: :get} do |f| %>
```
Comment (biznezo) [inline app/views/admin/users/index.html.erb]: insert indent after html

File: app/controllers/admin/users_controller.rb [hunk 9]
```diff
@@ -2,7 +2,13 @@ class Admin::UsersController < ApplicationController
   load_and_authorize_resource
 
   def index
-    @users = @users.paginate page: params[:page]
+    @search = User.search(params[:q])
```
Comment (biznezo) [inline app/controllers/admin/users_controller.rb]: ()

**Pull request merged**